xgcm/test/test_integrate.py

```diff
@@ -29,43 +33,105 @@ def test_integrate(gridtype):
-            ["X", "Y", ["X", "Y"]],
-            ["dx_e", "dy_e", "area_e"],  # need more metrics?
-            ["xu", "yt", ["xu", "yt"]],
-        ):
-            integrated = grid.integrate(ds.u, axis)
-            expected = (ds.u * ds[metric_name]).sum(dim)
+    # test v position
+    for axis, metric_name, dim in zip(
+        ["X", "Y", ["X", "Y"]],
+        ["dx_ne", "dy_ne", "area_ne"],  # need more metrics?
+        ["xu", "yu", ["xu", "yu"]],
+    ):
+        integrated = grid.integrate(ds.v, axis)
+        expected = (ds.v * ds[metric_name]).sum(dim)
+        assert integrated.equals(expected)
+
+
+def test_integrate_cgrid():
+    ds, coords, metrics = _patch_dataset("C")
+    grid = Grid(ds, coords=coords, metrics=metrics)
+    # test tracer position
+    for axis, metric_name, dim in zip(
+        ["X", "Y", "Z", ["X", "Y"], ["X", "Y", "Z"]],
+        ["dx_t", "dy_t", "dz_t", "area_t", "volume_t"],
+        ["xt", "yt", "zt", ["xt", "yt"], ["xt", "yt", "zt"]],
+    ):
+        integrated = grid.integrate(ds.tracer, axis)
+        expected = (ds.tracer * ds[metric_name]).sum(dim)
+        assert integrated.equals(expected)
+        # test with tuple input if list is provided
+        if isinstance(axis, list):
+            integrated = grid.integrate(ds.tracer, tuple(axis))
+            expected = (ds.tracer * ds[metric_name]).sum(dim)
             assert integrated.equals(expected)
 
+    # test u positon
+    for axis, metric_name, dim in zip(
+        ["X", "Y", ["X", "Y"]],
+        ["dx_e", "dy_e", "area_e"],  # need more metrics?
+        ["xu", "yt", ["xu", "yt"]],
+    ):
+        integrated = grid.integrate(ds.u, axis)
+        expected = (ds.u * ds[metric_name]).sum(dim)
+        assert integrated.equals(expected)
+
     # test v positon
-    if gridtype == "B":
-        for axis, metric_name, dim in zip(
-            ["X", "Y", ["X", "Y"]],
-            ["dx_ne", "dy_ne", "area_ne"],  # need more metrics?
-            ["xu", "yu", ["xu", "yu"]],
-        ):
-            integrated = grid.integrate(ds.v, axis)
-            expected = (ds.v * ds[metric_name]).sum(dim)
-            assert integrated.equals(expected)
-    elif gridtype == "C":
-        for axis, metric_name, dim in zip(
-            ["X", "Y", ["X", "Y"]],
-            ["dx_n", "dy_n", "area_n"],  # need more metrics?
-            ["xt", "yu", ["xt", "yu"]],
-        ):
-            integrated = grid.integrate(ds.v, axis)
-            expected = (ds.v * ds[metric_name]).sum(dim)
-            assert integrated.equals(expected)
+    for axis, metric_name, dim in zip(
+        ["X", "Y", ["X", "Y"]],
+        ["dx_n", "dy_n", "area_n"],  # need more metrics?
+        ["xt", "yu", ["xt", "yu"]],
+    ):
+        integrated = grid.integrate(ds.v, axis)
+        expected = (ds.v * ds[metric_name]).sum(dim)
+        assert integrated.equals(expected)
+
+
+@pytest.mark.parametrize("axis", ["X", "Y", "Z"])
+def test_integrate_missingaxis(axis):
```

Comment:
So I am testing two error cases: 
1) An axis specified in `integrate` missing in the grid object
2) A missing dimension  in the input dataarray (e.g. because it was sliced or averaged)

Can you think of anything else we want to catch here?

Reply:
Nothing else comes to mind.